### ISSUE TITLE
Update the OFL to be without reserved font name

### ISSUE THREAD

**Issue by andreaslarsen** [fontforge/fontforge#2184]
Since it's the default for OFL 1.1 shouldn't it be removed from the license text (users wanting reserved font names can add it them themselves)

I'll update https://github.com/fontforge/fontforge/blob/519c5f74a72424125717769fb81a11069f7a3bc5/fontforge/ofl.c if you guys think it's a good idea.

**Comment by davelab6** [comment]
Great idea :)

**Comment by n7s** [comment]
The default? For whom exactly? According to which policy? 

This is what the original OFL.txt example header actually has:  

Copyright (c) <dates>, <Copyright Holder> (<URL|email>),
with Reserved Font Name <Reserved Font Name>.
Copyright (c) <dates>, <additional Copyright Holder> (<URL|email>),
with Reserved Font Name <additional Reserved Font Name>.
Copyright (c) <dates>, <additional Copyright Holder> (<URL|email>

Showing the original header and leaving designers to decide for themselves is probably a better approach.

You should indicate that it's an optional feature and point to the relevant FAQ entry (http://scripts.sil.org/OFL-FAQ_web#cbd32a36 ) but going out of your way to rewrite the original header and hide that feature is really making a decision instead of the authors.

**Comment by andreaslarsen** [comment]
Having the RFN text in the header as default is just as much making a decision as not having it. There are basically two users: 

1. those who know about RFN - they will know how to add it themselves
2. those who don't - they will think it's a mandatory part of the license

In other words:

a. Fontforge with RFN as default => many new users will keep it and has to be told to remove it later on to allow for e.g. simple webfont conversions without the need for renaming

a. Fontforge without RFN as default => advanced users will know how to re add it and new user will just release fonts without RFN => no consequences (at least I haven't heard of a single case of font name hijacking but I might be wrong)

The flaw is obviously in OFL where a license with the same name can be with and without RFN - they should have given one of the licenses a suffix similar to how cc licenses are made. Or even better: no RFN at all - it's a hypothetical scenario anyways.

**Comment by n7s** [comment]
Well, the recommended but optional reserved naming system was a key aspect of the license from the beginning. It's far from simple, have you read http://scripts.sil.org/OFL_web_fonts_and_RFNs about the importance of functional equivalence for webfonts? There are differing opinions on naming. There is certainly room to clarify what is meant and let users decide with all the facts. Even for libre/open fonts with generous commissioning, I notice various authors choose to retain a RFN and make use of that feature in the license. There must be a reason, a value attached to control of the canonical name by the original author. Maybe it's because some conversions and optimization actually break the features of the font? That bit of code in fontforge goes back a few years, when webfonts were a very new thing. AFAICT it was to encourage designers to actually fill in their headers with their own information with these placeholders.  IMHO, retrofitting views about what users should be doing with an existing model is probably not the best approach. Explaining the hows and why would be better.  If you want another license without that feature at all, there are others to pick from :-)

**Comment by davelab6** [comment]
A lot of people I commissioned fonts from applied an RFN because it was
default in the template, and remove it when given the whys. Eg
https://github.com/clauseggers/Inknut-Antiqua/pull/2

**Comment by andreaslarsen** [comment]
@davelab6 yep, I was one of said people too ;) 
@n7s explaining is sometimes better but here it just adds to the confusion as a beginner in the typographic world and adds "clutter" to a program that has a steep learning curve already. 

In my opinion the whole concept of the need for RFN is flawed. Yes conversion and optimisations can make a font look worse but why would we as font designers of free fonts care? Bad design choices, css antialiasing "tricks" etc can make it look a lot worse...

**Comment by andreaslarsen** [comment]
@n7s what licenses are you thinking about - the great thing about OFL is how it specifies how the font can never be resold alone while still allowing users to modify, contribute etc. as they want

**Comment by khaledhosny** [comment]
To me, OFL has two selling points 1) it is copyleft 2) other libre fonts are using it, otherwise I find it a poor license trying to satisfy everyone but actually satisfying none. RFN should die (together with the silly no-sell-alone thing, but that one is harder to drop, without an explicit “exception” at least).

**Comment by andreaslarsen** [comment]
@khaledhosny why not just use MIT if you don't care for the no-sell-alone part?

**Comment by davelab6** [comment]
See his (1): copyleft

**Comment by khaledhosny** [comment]
What Dave said. I don’t mind people selling my work (or making profit of any other kind), but I do mind people making non-libre derivatives of my work so I use copyleft licenses whenever it is possible. I would have used GPL if it weren’t for the fact that few libre fonts are using it (some think GPL does not work well for fonts, but I don’t think there is really anything major); being able to reuse and remix libre fonts is much so invaluable  that it outweighs any shortcomings I might see with OFL.

**Comment by adrientetar** [comment]
> together with the silly no-sell-alone thing

I watched the talk of Hrant Papazian where he says he prefers Apache license so font developers can make money out of character set extensions.  
Personally, I'd just use the SIL but also add a note in the README that if a developer wants to make a paid extension to my family, he can just contact me and I relicense it solely for said project.

**Comment by davelab6** [comment]
On 9 April 2015 at 10:45, Adrien Tétar <notifications@github.com> wrote:

> I watched the talk of Hrant Papazian where he says he prefers Apache
> license so font developers can make money out of character set extensions.


He's talking complete nonsense! I'm sorry I didn't make time to post a full
rebuttal to him yet.

**Comment by davelab6** [comment]
The short version is: Proprietary licensing developers do not do this with
the permissively licensed fonts already published, and they have had
literally YEARS to do so. If you ask font developers if they are interested
in such work, they say, "no! I'd much rather work on fonts that are totally
my own." The funding for the kind of projects Hrant described tends to come
from institutions, not speculation on the proprietary retail market, and
those institutions with small budgets that can't afford a new Latin
generally want to make the fonts widely available to their language
communities, so they want them to be all libre - so the OFL is fine.

**Comment by adrientetar** [comment]
I do not disagree, I'm just saying that you can use the SIL without closing the door to eventual arrangements.

**Comment by andreaslarsen** [comment]
I just spent five minutes, mostly for fun, trying to "shave" down the OFL to the copyleft and no-sell-alone only. I hate long licenses and wished something likes this was available for fonts:

https://gist.github.com/andreaslarsen/a702f896a18550f505d5

**Comment by davelab6** [comment]
I don't think this is a good idea :) OFL is short enough :)
http://google-opensource.blogspot.com/2008/05/standing-against-license-proliferation.html

**Comment by andreaslarsen** [comment]
I know there are too many licenses already but OFL is by no means short and the fact that their faq is 5km long shows how flawed it is;)

**Comment by khaledhosny** [comment]
> http://google-opensource.blogspot.com/2008/05/standing-against-license-proliferation.html

Can’t agree more.

**Comment by andreaslarsen** [comment]
Why not include say the GPL3 as default instead of OFL?

**Comment by davelab6** [comment]
Because its source provision requirement is a pain, especially for fonts

**Comment by andreaslarsen** [comment]
Fair enough and sorry for asking all the noob questions - I just think OFL stinks but haven't found anything better:(

**Comment by davelab6** [comment]
You are very welcome! :) 

I think this is why GPL is not favoured by the latest generation of developers; its just too much to bother with, and a simple permissive license is hassle free. 

I agree OFL could be improved, but SIL has no resources to do a round of revision, and they (IMHO unwisely) didn't include an automatic upgrade clause like Creative Commons has (and the FSF doesn't include but recommends) so we are stuck with OFL v1.1 until some disaster happens, which seems unlikely.

**Comment by adrientetar** [comment]
I think that the SIL OFL works well for its purposes personally.

**Comment by andreaslarsen** [comment]
I works well - it's just not as easily understandable by newcomers and end users as most other open licenses. But sorry for taking my own issue off topic... 

On topic: I really don't see why Fontforge shouldn't either remove RFN or at least emphasise that it's optional and maybe tell why it's a bad idea in most cases.

**Comment by adrientetar** [comment]
> I really don't see why Fontforge shouldn't either remove RFN or at least emphasise that it's optional and maybe tell why it's a bad idea in most cases.

Agreed, it's to be done still.

**Comment by davelab6** [comment]
> emphasise that it's optional 

I think that would be best.
